lib/iris/_merge.py

```diff
@@ -408,8 +408,10 @@ def match(self, other, error_on_mismatch):
             msg = 'cube data dtype differs: {} != {}'
             msgs.append(msg.format(self.data_type, other.data_type))
         if self.fill_value != other.fill_value:
-            msg = 'cube fill value differs: {} != {}'
-            msgs.append(msg.format(self.fill_value, other.fill_value))
+            # Allow fill-value promotion if either fill-value is None.
+            if self.fill_value is not None and other.fill_value is not None:
+                msg = 'cube fill value differs: {} != {}'
+                msgs.append(msg.format(self.fill_value, other.fill_value))
```

Comment:
@dkillick So we only fail on `fill_value` if they are both non-`None` and not equal. Otherwise, the `fill_value`s are either the same, or either `fill_value` is `None`, which is the promotion case.